unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/ChannelEntryFactory.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace DCL.Chat.HUD
+{
+    [CreateAssetMenu(fileName = "ChannelEntryFactory", menuName = "DCL/Social/ChannelEntryFactory")]
+    public class ChannelEntryFactory : ScriptableObject
+    {
+        [Serializable]
+        private struct SpecialEntry
+        {
+            public string channelId;
+            public PublicChatEntry prefab;
+        }
+        
+        [SerializeField] private PublicChatEntry defaultEntryPrefab;
+        [SerializeField] private SpecialEntry[] specialEntryPrefabs;
+
+        private Dictionary<string, PublicChatEntry> specialEntries;
+
+        public PublicChatEntry Create(string channelId)
+        {
+            specialEntries ??= specialEntryPrefabs.ToDictionary(entry => entry.channelId, entry => entry.prefab);
```

Comment:
As `specialEntries` lifetime will persist between switching the play mode in editor, it can happen that `specialEntries` can have old data if we modify the scriptable object. Do we have any pattern for solving that? Or we are ok with that corner case?

Reply:
i wouldn't worry about that really